proxy/src/telemetry/metrics/serve.rs

```diff
@@ -63,22 +77,73 @@ impl Service for Serve {
             let mut writer = GzEncoder::new(Vec::<u8>::new(), CompressionOptions::fast());
             write!(&mut writer, "{}", *metrics)
                 .and_then(|_| writer.finish())
-                .map(|body| {
-                    Response::new()
-                        .with_header(ContentEncoding(vec![Encoding::Gzip]))
-                        .with_header(ContentType::plaintext())
-                        .with_body(Body::from(body))
+                .map_err(ServeError::from)
```

Comment:
I realize this behavior isn't a change, but it'd be better to use `or_else` and return a `Response` with a 500, instead of an actual error.

Reply:
I believe the function _will_ return 500 if this fails: https://github.com/runconduit/conduit/pull/944/files/21ce0747ee0f0708ecf90d503ad243b504c63933#r189123771